src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts

```diff
@@ -613,138 +644,276 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
+  }
 
-    this.refreshPinnedGenes();
+  ensgExistsInProteins(ensemblGeneId: string) {
+    const ensemblIndex = this.pinnedItems.findIndex(
+      (g: GCTGene) => g.ensembl_gene_id === ensemblGeneId
+    );
+    if (ensemblIndex < 0) {
+      return false;
+    }
+    return true;
+  }
+
+  pinProteins(proteins: GCTGene[]) {
+    let remaining = this.maxPinnedGenes - this.uniquePinnedGenesCount;
+    
+    let proteinsAdded = 0;
+
+    let showToast = false;
+    for (let i = 0; i < proteins.length; i++) {
+      // if remaining count is zero, show alert toast
+      if (remaining <= 0) {
+        // check border condition: when there are no remaining ensg slots, it is still possible there
+        // are proteins that could be added
+        showToast = true;
+        if (this.ensgExistsInProteins(proteins[i].ensembl_gene_id)) {
+          // if the gene exists, we can still add the protein
+          this.pinGene(proteins[i], false);
+          proteinsAdded++;
+          remaining = this.maxPinnedGenes - this.getCountOfUniqueGenes();
+        }
+      } else {
+        // add protein to pinned collection
+        this.pinGene(proteins[i], false);
+        proteinsAdded++;
+        // have to call method below since we need to recompute the count of unique genes
+        remaining = this.maxPinnedGenes - this.getCountOfUniqueGenes();
+      }
+    }
+    if (showToast) {
+      this.showMaxPinnedRowsErrorToast(proteinsAdded);
+    }
+    this.uniquePinnedGenesCount = this.getCountOfUniqueGenes();
   }
 
-  unpinGene(gene: GCTGene, refresh = true) {
-    const index = this.pinnedGenes.findIndex(
+  onUnPinGeneClick(gene: GCTGene, refresh = true) {
+    this.setLastPinnedCategories();
+
+    const index = this.pinnedItems.findIndex(
       (g: GCTGene) => g.uid === gene.uid
     );
 
     if (index === -1) {
       return;
     }
 
-    this.pinnedGenes.splice(index, 1);
+    this.pinnedItems.splice(index, 1);
 
     if (refresh) {
       this.clearPinnedGenesCache();
       this.refreshPinnedGenes();
     }
+
+    if (this.category === 'Protein - Differential Expression')
+      this.uniquePinnedGenesCount = this.getCountOfUniqueGenes();
+  }
+
+  onClearAllClick() {
+    this.setLastPinnedCategories();
+    this.clearPinnedGenes();
   }
 
   clearPinnedGenes() {
-    this.pinnedGenes = [];
+    this.pinnedItems = [];
     this.clearPinnedGenesCache();
     this.refreshPinnedGenes();
   }
 
   getPinnedEnsemblGeneIds() {
-    return this.pinnedGenes.map((g: GCTGene) => g.ensembl_gene_id);
+    return this.pinnedItems.map((g: GCTGene) => g.ensembl_gene_id);
   }
 
   getPinnedUniProtIds() {
-    return this.pinnedGenes.map((g: GCTGene) => g.uniprotid);
+    return this.pinnedItems.map((g: GCTGene) => g.uniprotid);
+  }
+
+  getPinDisabledStatus() {
+    if (this.category === 'RNA - Differential Expression')
+      return this.pinnedItems.length >= this.maxPinnedGenes;
+    else {
+      // default to showing pin all button for protein view
```

Comment:
I think`getPinDisabledStatus` is used for both individual item pins and pin all button, so could the comment be generalized? 

```suggestion
      // default to showing pin buttons in protein view
```

Reply:
done!